submissions/webdevtoday/friends-app/js/app.js

```diff
@@ -0,0 +1,95 @@
+import { getUsers } from './components/RandomUserAPI.js';
+import { normalizeUsers } from './components/normalizeUsers.js';
+import { makeCard } from './components/makeCard.js';
+import { oops } from './components/oops.js';
+import { makeForm } from './components/makeForm.js';
+import { sortAndFiltersFunctions } from './components/sortAndFiltersFunctions.js';
+
+let users = [];
+const sortAndFiltersState = {
+    // filters:
+    age: null,
+    name: null,
+    gender: null,
+
+    // sorts:
+    sort: null,
+};
+
+const sidebar = document.getElementsByClassName('sidebar')[0];
+const sidebarOpener = document.getElementsByClassName('sidebar-opener')[0];
+const sidebar__container = document.getElementsByClassName('sidebar__container')[0];
+const cards = document.getElementsByClassName('cards')[0];
```

Comment:
Don't use constructions like this, or children[0], firstElementChild, parentNode, nextSibling, etc. In such way, you rely on the order of DOM elements. So in case when you will have changed design - your code will be broken. Which is bad, obviously. Use querySelector.

Reply:
I'll change it, but I don't get you about it. When we use querySelector it finds first element by query. And getElementsByClassName(class)[0] does the same.

Reply:
I've done it.